metaflow/plugins/cards/card_creator.py

```diff
@@ -0,0 +1,213 @@
+import time
+import subprocess
+import tempfile
+import json
+import sys
+import os
+from metaflow import current
+
+ASYNC_TIMEOUT = 30
+
+
+class CardProcessManager:
+    """
+    This class is responsible for managing the card creation processes.
+
+    """
+
+    async_card_processes = {
+        # "carduuid": {
+        #     "proc": subprocess.Popen,
+        #     "started": time.time()
+        # }
+    }
+
+    @classmethod
+    def _register_card_process(cls, carduuid, proc):
+        cls.async_card_processes[carduuid] = {
+            "proc": proc,
+            "started": time.time(),
+        }
+
+    @classmethod
+    def _get_card_process(cls, carduuid):
+        proc_dict = cls.async_card_processes.get(carduuid, None)
+        if proc_dict is not None:
+            return proc_dict["proc"], proc_dict["started"]
+        return None, None
+
+    @classmethod
+    def _remove_card_process(cls, carduuid):
+        if carduuid in cls.async_card_processes:
+            cls.async_card_processes[carduuid]["proc"].kill()
+            del cls.async_card_processes[carduuid]
+
+
+class CardCreator:
+    def __init__(self, top_level_options):
+        self._top_level_options = top_level_options
+
+    def create(
```

Comment:
Given this is user-driven, it is possible that things get out of hand if, for example, the user doesn't call `refresh` for a while -- subprocesses can go crazy and have free rein to wreak havoc.

Reply:
I mean you check for subprocess timeouts in a user-driven manner (ie: there is no separate thing that checks periodically). So if the user does one refresh but then nothing else, the subprocess timeout isn't checked. At least I think that's what I remember from my first review.